tokio-quiche/src/quic/router/mod.rs

```diff
@@ -524,8 +547,20 @@ where
                                 ControlMessageOwned::Ipv6PacketInfo(_) => {
                                     // We only want the destination address from
                                     // IP_RECVORIGDSTADDR, but we'll get these
-                                    // messages because
-                                    // we set IP_PKTINFO on the socket.
+                                    // messages because we set IP_PKTINFO on the
+                                    // socket.
+                                },
+                                ControlMessageOwned::Unknown(raw_cmsg) => {
+                                    let UnknownCmsg {
+                                        cmsg_header,
+                                        data_bytes,
+                                    } = raw_cmsg;
+
+                                    if cmsg_header.cmsg_level == SOL_SOCKET &&
+                                        cmsg_header.cmsg_type == SO_MARK
+                                    {
+                                        let _ = mark_bytes.insert(data_bytes);
+                                    }
```

Comment:
Do we need to validate the size of `data_bytes`. Since this grows the `reusable_cmsg_space` on a socket, and since the socket is reused, this might be important.

If we decide to restrict size, does it make sense to make this a static slice?

Reply:
spoke offline. The SO_MARK cmsg will be marked by the kernel and is expected to be 32 bits.

Reply:
I can change it to a `[u8; 4]` for ergonomics in userspace, since we know that `SO_MARK` is a u32